cli/cook/subcommands/submit.py

```diff
@@ -0,0 +1,191 @@
+
+
+def safe_pop(d, key):
+    """If key is present in d, pops and returns the value. Otherwise, returns None."""
+    value = d.pop(key) if key in d else None
+    return value
+
+
+def read_commands_from_stdin():
+    """Prompts for and then reads subcommands, one per line, from stdin"""
+    print_info('Enter the subcommands, one per line (press Ctrl+D on a blank line to submit)')
+    commands = read_lines()
+    if len(commands) < 1:
+        raise Exception('You must specify at least one command')
+    return commands
+
+
+def read_jobs_from_stdin():
+    """Prompts for and then reads job(s) JSON from stdin"""
+    print('Enter the raw job(s) JSON (press Ctrl+D on a blank line to submit)', file=sys.stderr)
+    jobs_json = sys.stdin.read()
+    return jobs_json
+
+
+def submit(clusters, args):
+    """
+    Submits a job (or multiple jobs) to cook scheduler. Assembles a list of jobs,
+    potentially getting data from configuration, the command line, and stdin.
+    """
+    logging.debug('submit args: %s' % args)
+
+    job = args
+    raw = safe_pop(job, 'raw')
+    command_from_command_line = safe_pop(job, 'command')
+    command_args = safe_pop(job, 'args')
+
+    if raw:
+        if command_from_command_line:
+            raise Exception('You cannot specify a command at the command line when using --raw/-r')
+
+        jobs_json = read_jobs_from_stdin()
+        jobs = parse_raw_job_spec(job, jobs_json)
+    else:
+        if command_from_command_line:
+            commands = ['%s%s' % (command_from_command_line, (' ' + ' '.join(command_args)) if command_args else '')]
+        else:
+            commands = read_commands_from_stdin()
+
+        logging.debug('subcommands: %s' % commands)
+
+        if job.get('uuid') and len(commands) > 1:
+            raise Exception('You cannot specify multiple subcommands with a single UUID')
+
+        if job.get('env'):
+            job['env'] = {e.split('=')[0]: e.split('=')[1] for e in job['env']}
+
+        jobs = [merge(job, {'command': c}) for c in commands]
+
+    for j in jobs:
+        if not j.get('uuid'):
+            j['uuid'] = str(uuid.uuid4())
+
+        if not j.get('name'):
+            j['name'] = '%s_job' % os.environ['USER']
+
+    return submit_federated(clusters, jobs)
+
+
+def valid_uuid(s):
+    """Allows argparse to flag user-provided job uuids as valid or not"""
+    try:
+        return str(uuid.UUID(s))
+    except:
+        raise argparse.ArgumentTypeError('%s is not a valid UUID' % s)
+
+
+def register(add_parser):
+    """Adds this sub-command's parser and returns the action function"""
+    submit_parser = add_parser('submit', help='create job for command')
+    submit_parser.add_argument('--uuid', '-u', help='uuid of job', type=valid_uuid)
+    submit_parser.add_argument('--name', '-n', help='name of job')
+    submit_parser.add_argument('--priority', '-p', help='priority of job, between 0 and 100 (inclusive)',
```

Comment:
Should probably clarify if 0 or 100 is the "highest" priority. Mentioning the default might be helpful too.

Reply:
Good idea, done